en/docs/learn/design-api/publish-api/publish-an-api-with-artifact-synchronization.md

```diff
@@ -0,0 +1,158 @@
+artifact_retriever = "DBRetriever"
+deployment_retry_duartion = 15000
+data_retrieval_mode = "sync"
+save_artifacts_locally = false
+```
+
+ - Through artifact_retriever we can specify the extension point. The default is `DBRetriever` where the artifacts are
+  pulled from the database.
+  
+ - In gateway_labels we can specify the labels which the gateway is going to subscribe to. Only the APIs with these
+  labels will be pulled from the extension point and deployed.
+  
+ - If we add the config `save_artifacts_locally = true` or remove `save_artifacts_locally` config  then synapse
+  artifacts will be stored in the file system. (Saved in
+   `<APIM_HOME>/repository/deployment/server/synapse-configs/default/`) . When `save_artifacts_locally = false`  
+   then the artifacts from extension point will not be stored in the file system.
+   
+ - Through `data_retrieval_mode = “sync” ` we can specify the mode of deployment of artifacts from the extension point
+ . By default gateway Startup will be in a Synchronous manner. Here the server will wait until all the APIs have been
+  deployed. If there is any failure in the deployment, it will again be triggered for n times where n is the maximum
+   retry count. If the APIs are not deployed even in n retries then the server will start with un deployed API artifacts.
+   If the user wants to switch the startup mode to an Asynchronous mode then he needs to specify it in the configs as
+  `data_retrieval_mode = "async"`. Here the server will be up, independent of the deployment of synapse artifacts in
+   gateway.
+   
+ - We can specify the retry duration in milliseconds to deploy artifacts, if there is a failure in pulling them from the
+  extension through `deployment_retry_duartion`. The retry duration specified here will be exponentially increased by a 
+  progression factor of 2. That means the duration will be progressed as 15s, 30s, 60s, 120s……. if there are continuous
+  failures. And retry duration is bounded with 1 hr.
+  
+  <html><div class="admonition note">
+  <p class="admonition-title">Note</p>
+  <p> This `deployment_retry_duartion` is  applicable only in Asynchronous deployment
+  ( `data_retrieval_mode = "async"` ) 
+   where the server will try to deploy the artifacts after it is started. </p>
+  </div>
+  </html>
+   
+```
+[apim.event_hub] 
+enable = true
+service_url = "https://traffic-manager:9443"
+```
+
+ - If you are not running gateway as an all in one pack and it runs with a port offset, or when it's in a different
+  node, we have to specify event_hub service_url as well .
+
+###Database Configurations
+
+By default, the data related to synapse artifacts will be stored in WSO2AM_DB. But you can specify a separate database 
+for this feature. Only the data related to the synapse artifacts will be saved in this database.
+
+First you need to add a configuration element of `[database.sync_runtime_artifacts_db]` . And simply have to update
```

Comment:
-  Open the <API-M_HOME>/repository/conf/deployment.toml file and add the following configuration.
``` tab="Format"
[database.sync_runtime_artifacts_db]
type = "mysql"
url = "jdbc:mysql://localhost:3306/<DATABASE_NAME>"
username = "<USER_NAME>"
password = "<PASSWORD>"
```

``` tab="Example"
[database.sync_runtime_artifacts_db]
type = "mysql"
url = "jdbc:mysql://localhost:3306/WSO2AS_DB"
username = "wso2carbon"
password = "wso2carbon"
```
  | Element                       | Description                                                 |
    |-------------------------------|-------------------------------------------------------------|
    | `type`                      | The database type used                                      |
    | `url`                       | The URL of the database. The default port for MySQL is 3306 |
    | `username` and `password` | The name and password of the database user                  |
    | `driverClassName`          | The class name of the database driver                       |